src/runtime/Types/ClassDerived.cs

```diff
@@ -678,6 +710,127 @@ private static ModuleBuilder GetModuleBuilder(string assemblyName, string module
 
             return moduleBuilder;
         }
+
+        /// <summary>
+        /// Builds a C# attribute based on a python tuple definition (Attibute Type, *attributeArgs, ** attributeKwArgs)
+        /// </summary>
+        static CustomAttributeBuilder GetAttributeBuilder(PyObject attr)
+        {
+            // this is a bit complicated because we want to support both unnamed and named arguments
+            // in C# there is a discrepancy between property setters and named argument wrt attributes
+            // in python there is no difference. But luckily there is rarely a conflict, so we can treat
+            // named arguments and named properties or fields the same way.
+            var tp = new PyTuple(attr);
+            Type attribute = tp[0].As<Type>();
```

Comment:
Same as below, you need to handle the scenario where the object passed here is not actually a `Type` properly.